### DISCUSSION TITLE
How to get the attribute value in the item type by using ECSql

### DISCUSSION THREAD

**Discussion by JingWang111** [iTwin/itwinjs-core#5572] [Q&A] [answered]
I want to get the attribute value in the item type by using Ecsql
I've already defined the item type, and attach property values to element
![image](https://github.com/iTwin/itwinjs-core/assets/48142967/d053d994-4aad-48f4-8512-221b88f6672c)
![MM_4HZSI7KY%HXPYBGQUSO6](https://github.com/iTwin/itwinjs-core/assets/48142967/6acbcc0d-db22-49b3-abcb-b43e4a1b30a8)

**Comment by pratikshan85** [comment]
@JingWang111 Can you please give us access to the iModel on which you are trying to run ECSql? 
You can also give us the input design file for our detailed analysis.
Contact email : pratikshan.ghosh@bentley.com

**Reply by JingWang111** [reply]
Thanks for your reply.
I have sent bim and dgn files to your email and look forward to your reply

**Comment by pratikshan85** [comment, marked as answer]
@JingWang111 Please find the below query :

**select cd.progressCode from DgnCustomItemTypes_Road.CodeElementAspect cd JOIN bis.ElementOwnsMultiAspects ml ON cd.ECInstanceId = ml.TargetECInstanceId JOIN bis.Element el ON el.ECInstanceId = ml.SourceECInstanceId where el.UserLabel='Extended Element'**

**Note : I have also separately emailed you for your test model.**

**Reply by JingWang111** [reply]
Thank you very much for your reply to solve my problem, but I found another problem. 
When I converted the dgn file to the bim file through snapshot based on OBD-bridge, OBD-bridge not the MS-birdge, I found that I could not find the property value I wanted with this ECSQL, not even the DgnCustomItemTypes-xxx table.  
I have sent bim and dgn files to your email and look forward to your reply.

**Reply by pratikshan85** [reply]
@JingWang111 

After the conversion of the dgn file to iModel your custom file name has changed to :
DgnCustomItemTypes_PHK. CodeElementAspect 
DgnCustomItemTypes_PHK. Code2ElementAspect

Please find the below ECSql query :

**select cd.progressCode2,cd.qualityCode2 from DgnCustomItemTypes_PHK. Code2ElementAspect cd JOIN bis.ElementOwnsMultiAspects ml ON cd.ECInstanceId = ml.TargetECInstanceId JOIN bis.Element el ON el.ECInstanceId = ml.SourceECInstanceId where el.UserLabel='Extended Element'**

**select cd.progressCode,cd.qualityCode from DgnCustomItemTypes_PHK. CodeElementAspect cd JOIN bis.ElementOwnsMultiAspects ml ON cd.ECInstanceId = ml.TargetECInstanceId JOIN bis.Element el ON el.ECInstanceId = ml.SourceECInstanceId where el.UserLabel='Extended Element'**

**Reply by JingWang111** [reply]
Thank you very much for your reply. I have tried the method you mentioned, and I found one  problem.

The problem is that if item-type information is written in Chinese, DgnCustomItemTypes_xxx from the bim file can get the name like this DgnCustomItemTypes___x7F16____x7801__ , but "select * from DgnCustomItemTypes___x7F16____x7801__.CodeElementAspect "is empty, the table does not exist, May I ask how to deal with the itemtype written in Chinese data?

I have sent bim and dgn files to your email and look forward to your reply.

**Reply by pratikshan85** [reply]
Hello,
As you are creating a new dgn file with a custom class internally it will create new class name.
I would prefer you to please take reference of the ECSql I have shared with you in the earlier case.
Also, Please navigate the Schema browser once you create the iModel and then you can identify the newly created custom class.
1.	Please open your project & iModel in https://imodelconsole.bentley.com/.
2.	Top right corner in the settings icon you will find “Schema Explorer” . Please visit the link and explore the schema.

**Reply by JingWang111** [reply]
Thank you, I will try this out.
